hook.js

```diff
@@ -180,7 +180,7 @@ async function processModule ({
       // needs to utilize that new name while being initialized from the
       // corresponding origin namespace.
       const renamedExport = matches[2]
-      setters.set(`$${renamedExport}${ns}`, `
+      setters.set(`$${renamedExport}`, `
```

Comment:
Does this change invalidate the preceding comment, which is saying:

1. If `a.mjs` exports entities from `b.mjs`
2. And `b.mjs` exports a default entity
3. The default entity exported by `b.mjs` should be named `b` under the namespace associated with `b.mjs`

Reply:
Not sure, need to check that the tests cover this!

Reply:
Why does iitm have this renaming of default exports behaviour? It means that iitm adds additional exports that don't exist when loader isn't in use. Why would this ever be desirable?

Reply:
> Why does iitm have this namespacing of default exports behaviour?

IITM basically has to re-implement ESM loading from scratch because it needs to be able to wrap up any of the entities exported by a module. But transitive modules can export entities with the same identifiers as the module that loaded the transitive dependency (or any ancestor module for that matter). This, combined with star exports, means we have to figure out which one is which in some sort of way. The way chosen is a namespace per module.

1. https://github.com/DataDog/import-in-the-middle/pull/43 added support for handling star exports
2. https://github.com/DataDog/import-in-the-middle/pull/53 attempts to fix the duplicates exports made possible by 1

All of this could be avoided if we had an API where Node does all of the ESM loading and merely gives us back a malleable set of exports.